### ISSUE TITLE
After update error, ArgumentNullException: Value cannot be null. (Parameter 'provider')

### ISSUE THREAD

**Issue by WEBhicham** [umbraco/Umbraco-CMS#12643]
### Which *exact* Umbraco version are you using? For example: 9.0.1 - don't just write v9

10.0.1

### Bug summary

After updating my umbraco website from 10.0.0 to 10.0.1 to fix the redirect error I got the following error:
ArgumentNullException: Value cannot be null. (Parameter 'provider')

This issue is fixed after downgrading to 10.0.0

### Specifics

![image](https://user-images.githubusercontent.com/27362642/176645428-93183cb8-4ad2-4128-a3b8-46ccd1aa66fd.png)

### Steps to reproduce

Update from 10.0.0 to 10.0.1

### Expected result / actual result

# Expected
My website just runs

# Actual
I get an an error

**Comment by nikolajlauridsen** [comment]
Hey @WEBhicham, this seem to be caused by your `Program.cs` missing the call to `.ConfigureUmbracoDefaults()`, in the `CreateHostBuilder` method. That's the way I'm able to reproduce it. 

You can see it in more details if you look at step 8 in [this guide](https://our.umbraco.com/documentation/Fundamentals/Setup/Upgrading/version-specific#steps-to-upgrade-using-visual-studio), after you follow these steps the error should go away 😄

**Comment by PerplexDaniel** [comment]
We also ran into this issue and I can confirm the fix mentioned by @nikolajlauridsen resolves it. 
I did not actually expect a patch release to require any code changes though so that was a bit of a surprise 😅

**Comment by nikolajlauridsen** [comment]
Glad to hear that the fix resolved it for you 😄 

Of course it's not optimal that this only surfaced in a patch, but to be fair, the code change was required for the major release, the issue have just not surfaced for everyone until this patch. 

Since there should only be breaking changes in major, code change isn't required for patches 😄

**Comment by jrunestone** [comment]
This exact error now hit me, when upgrading from 10.0.0 to 10.1.0. I already had the `ConfigureUmbracoDefaults()` in my `Program.cs`.

**Comment by jrunestone** [comment]
Ok so for me, the following is causing the error.

In `Program.cs`, using the default syntax of lambda-style methods as shipped with the Umbraco templates, I can run 10.1.0 without issues. However, using the "regular" block-style method syntax I get the error above. As far as I know, there's no difference except for the syntax?

Here's the code with the syntax that won't work:
```
public class Program
    {
        public static void Main(string[] args)
        {
            CreateHostBuilder(args)
                .Build()
                .Run();
        }

        public static IHostBuilder CreateHostBuilder(string[] args)
        {
            var builder = Host.CreateDefaultBuilder(args);
    
            builder
                .ConfigureUmbracoDefaults()
                .ConfigureWebHostDefaults(webBuilder =>
                {
                    webBuilder.UseStaticWebAssets();
                    webBuilder.UseStartup<Startup>();
                });
    
            return builder;
        }
    }
```

**Comment by jrunestone** [comment]
...and the problem was that I'm returning the wrong builder object. Should return the builder from the last `webBuilder.UseStartup<Startup>();`
